frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/NonNativeExternalDataSourceConfiguration.tsx

```diff
@@ -0,0 +1,257 @@
+                                        PipelineStage.Source,
+                                        `managed-${item.source_id}`,
+                                        PipelineNodeTab.Schemas
+                                    )}
+                                >
+                                    {item.source_type} {item.source_prefix}
+                                </Link>
+                            )
+                        },
+                    },
+                    {
+                        key: 'prefix',
+                        title: 'Table',
+                        render: (_, item: any) => item.name,
+                    },
+                    {
+                        key: 'status',
+                        title: 'Status',
+                        width: 80,
+                        render: (_, item: any) => {
+                            const { isConfigured, message } = getTableStatus(item.id)
+                            const sourceMapping = sources_map?.[item.id]
+                            const hasAnyMapping = sourceMapping && Object.keys(sourceMapping).length > 0
+
+                            if (isConfigured) {
+                                return (
+                                    <Tooltip title={message}>
+                                        <div className="flex justify-center">
+                                            <IconCheck className="text-success text-lg" />
+                                        </div>
+                                    </Tooltip>
+                                )
+                            } else if (hasAnyMapping) {
+                                return (
+                                    <Tooltip title={message}>
+                                        <div className="flex justify-center">
+                                            <IconWarning className="text-warning text-lg" />
+                                        </div>
+                                    </Tooltip>
+                                )
+                            }
+                            return (
+                                <Tooltip title={message}>
+                                    <div className="flex justify-center">
+                                        <IconX className="text-muted text-lg" />
+                                    </div>
+                                </Tooltip>
+                            )
+                        },
+                    },
+                    ...Object.keys(MARKETING_ANALYTICS_SCHEMA).map((column) => ({
+                        key: column,
+                        title: `${column.replace(/_/g, ' ').replace(/\b\w/g, (l) => l.toUpperCase())}${
+                            MARKETING_ANALYTICS_SCHEMA[column].required ? ' (*)' : ''
+                        }`,
+                        render: (_: any, item: any) => renderColumnMappingDropdown(item, column),
+                    })),
```

Comment:
That's interesting, I now see why you built it that way, but I dont think that should live on `schema` though, maybe we can put it somewhere else?